app/controllers/books_controller.rb

```diff
@@ -0,0 +1,55 @@
+# Controller to handle book management
+class BooksController < ApplicationController
+  def show
+    book = Book.find_by(id: params[:id])
+    if book
+      render json: book, status: :ok
+    else
+      render json: nil, status: :not_found
+    end
+  end
+
+  def index
+    books = Book.search(params).page(params[:page]).per(params[:per_page])
+    render json: books, status: :ok
+  end
+
+  def create
+    book = Book.new(book_params)
+    authorize book
```

Comment:
this should be the first line, you don't need to pass the book, just use this authorize :book, what's important is that the user is an admin